### DISCUSSION TITLE
How to run two vercel-deployment-task with different vercelProjectId

### DISCUSSION THREAD

**Discussion by SomnathKadam** [vercel/community#3442] [Help]
### Summary

Hi Team, facing issue for running two `vercel-deployment-task` with different vercelProjectId, it only run on first task, second project task is not running it shows below message.

As per given below yaml file it only run on `devProjectID`

```
==============================================================================
Task         : Vercel Deployment Task
Description  : An Azure Pipelines Task Extension for automatically deploying to Vercel
Version      : 1.2.1
Author       : Vercel
Help         : https://github.com/vercel/vercel-azure-devops-extension
==============================================================================
Vercel Project Id specified by both `vercelProjectId` input and `VERCEL_PROJECT_ID` environment variable. Input field `vercelProjectId` will be used.
Vercel Org Id specified by both `vercelOrgId` input and `VERCEL_ORG_ID` environment variable. Input field `vercelOrgId` will be used.
Vercel Token specified by both `vercelToken` input and `VERCEL_TOKEN` environment variable. Input field `vercelToken` will be used.
.....
> No changes found.
Updated .vercel\.env.production.local file  [266ms]

> Downloading project settings
.....
```

And my yaml file is 

```
trigger:
- master

pool:
  name: Default

steps:
- task: Vercel.vercel-deployment-extension.vercel-deployment-task.vercel-deployment-task@1
  displayName: 'Deploying to Dev'
  inputs:
    vercelProjectId: '$(devProjectID)'
    vercelOrgId: '$(userID)'
    vercelToken: '$(token)'
    production: true
    debug: true
- task: Vercel.vercel-deployment-extension.vercel-deployment-task.vercel-deployment-task@1
  displayName: 'Deploying to Prod'
  inputs:
    vercelProjectId: '$(ProdProjectID)'
    vercelOrgId: '$(userID)'
    vercelToken: '$(token)'
    production: true
    debug: true

```

### Steps to Reproduce

```
trigger:
- master

pool:
  name: Default

steps:
- task: Vercel.vercel-deployment-extension.vercel-deployment-task.vercel-deployment-task@1
  displayName: 'Deploying to Dev'
  inputs:
    vercelProjectId: '$(devProjectID)'
    vercelOrgId: '$(userID)'
    vercelToken: '$(token)'
    production: true
    debug: true
- task: Vercel.vercel-deployment-extension.vercel-deployment-task.vercel-deployment-task@1
  displayName: 'Deploying to Prod'
  inputs:
    vercelProjectId: '$(ProdProjectID)'
    vercelOrgId: '$(userID)'
    vercelToken: '$(token)'
    production: true
    debug: true
```

**Comment by amyegan** [comment]
Hey @SomnathKadam. I want to be sure I understand your situation correctly. Do you have two separate projects, or are you trying to do a [production deployment](https://vercel.com/docs/concepts/deployments/environments#production) and a [preview deployment](https://vercel.com/docs/concepts/deployments/preview-deployments) for the same project?

**Reply by SomnathKadam** [reply]
Hi @amyegan, I have 2 diffenernt project in vercel dashboard

**Reply by jroho** [reply]
Hey @amyegan , I want to resurface this issue. We are running into the same problem. It's impossible to deploy to multiple projects using the Azure DevOps extension. The project id gets cached when running any follow-up tasks. See this example build log:

![image](https://github.com/vercel/community/assets/1479565/8987e66d-0aaa-40d8-b902-7c52202020c5)

**Reply by amyegan** [reply]
Thanks for reaching out about this. The extension was designed to be very naive. It’s essentially just a way to use the CLI more efficiently, so it doesn’t handle things like cleanup automatically. There’s probably a way to do this (just as there is with the CLI). It's just a matter of figuring out which steps are missing and configuring things to account for that.

**Reply by jroho** [reply]
I added a step in my foreach loop that purges the .vercel folder and clears the VERCEL_PROJECT_ID environment variable. Do you have any other ideas about what else the cleanup step may entail? Resolving this would remove a major limitation of the plugin. In our case, we need to deploy many projects, ideally within the same pipeline.

**Reply by jroho** [reply]
Hey @amyegan , I tried the following within my foreach loop and it seems to have done the trick:

```
      - script: |
          echo "##vso[task.setvariable variable=VERCEL_PROJECT_ID]"
```

Thanks for your time, I believe this is a valid work around for now.

**Reply by amyegan** [reply]
Thanks for sharing your workaround. I'm glad you got it working! 😄 

I'm not familiar enough with Azure DevOps to think of a different solution, but I'll pass this info along to the engineers who work on the extension to help with future development.
